tests/integration/components/frost-list-test.js

```diff
@@ -123,59 +123,284 @@ describe(test.label, function () {
+    beforeEach(function () {
+      const testItems = [
+        Ember.Object.create({id: '0'}),
+        Ember.Object.create({id: '1'})
+      ]
+      const testSelectedItems = [
+        Ember.Object.create({id: '0'})
+      ]
+
+      this.set('items', testItems)
+      this.set('selectedItems', testSelectedItems)
+      this.set('itemComparator', (lhs, rhs) => {
+        return lhs.get('id') === rhs.get('id')
+      })
+      this.render(hbs`
+        {{frost-list
+          item=(component 'frost-list-item')
+          hook='my-list'
+          items=items
+          selectedItems=selectedItems
+          itemComparator=itemComparator
+        }}
+      `)
+      return wait()
+    })
+
+    it('item 0 is selected', function () {
+      expect(this.$($hook('my-list-item-container', {index: 0})).hasClass('is-selected')).to.eql(true)
+    })
+    it('item 1 is not selected', function () {
+      expect(this.$($hook('my-list-item-container', {index: 1})).hasClass('is-selected')).to.eql(false)
+    })
+
+    describe('when itemComparator set back to default', function () {
+      beforeEach(function () {
+        this.render(hbs`
+        {{frost-list
+          item=(component 'frost-list-item')
+          hook='my-list'
+          items=items
+          selectedItems=selectedItems
+        }}
+      `)
+        return wait()
+      })
+      it('item 0 not selected ', function () {
+        expect(this.$($hook('my-list-item-container', {index: 0})).hasClass('is-selected')).to.eql(false)
+      })
+      it('item 1 not selected', function () {
+        expect(this.$($hook('my-list-item-container', {index: 1})).hasClass('is-selected')).to.eql(false)
+      })
+    })
+  })
+
+  describe('supports basic and specific click', function () {
     beforeEach(function () {
       const testItems = A([
-        {
-          id: '1',
-          isSelected: true
-        },
-        {
-          id: '2',
-          isSelected: false
-        }
+        Ember.Object.create({id: '0'}),
+        Ember.Object.create({id: '1'})
       ])
 
       this.set('items', testItems)
+      const testSelectedItems = A([])
+      this.set('selectedItems', testSelectedItems)
+      this.set('onSelectionChange', (selectedItems) => {
+        this.get('selectedItems').setObjects(selectedItems)
+      })
 
       this.render(hbs`
         {{frost-list
           item=(component 'frost-list-item')
           hook='my-list'
           items=items
+          selectedItems=selectedItems
+          onSelectionChange=onSelectionChange
         }}
       `)
+      return wait()
     })
 
-    it('selects pre selected item', function () {
-      return wait().then(() => {
-        expect(
-          this.$($hook('my-list-item-0')).hasClass('is-selected')
-        ).to.eql(true)
+    describe('when doing basic click', function () {
+      beforeEach(function () {
+        this.$(hook('my-list-item', {index: 0})).click()
```

Comment:
You should use `$hook()` when you want a jQuery object rather than wrapping it in `this.$()`.